grpc/pb-marshalling.go

```diff
@@ -322,7 +331,7 @@ func PBToAuthz(pb *corepb.Authorization) (core.Authorization, error) {
 }
 
 func newRegistrationValid(reg *corepb.Registration) bool {
-	return !(len(reg.Key) == 0 || len(reg.InitialIP) == 0 || reg.CreatedAt == 0)
+	return !(len(reg.Key) == 0 || len(reg.InitialIP) == 0)
```

Comment:
```suggestion
	return !(len(reg.Key) == 0 || len(reg.InitialIP) == 0 || reg.CreatedAt == 0)
```

I think we can keep this check as the `corepb.Registration` field `reg.CreatedAt` is still an `int64`. I feel I should also call out that while this method is called in 3 places our unit tests don't cover it.

Reply:
Thanks for taking a look! To explain a bit more what @aarongable is referring to (since I already wrote it up), the integration tests were failing with reg.CreatedAt because now the CreatedAt proto is actually zero as it should be before being set by sa.NewRegistration.

The check as it currently exists only doesn't fail tests because `CreatedAt` ends up being a garbage non-zero value after converted to proto as demonstrated by `b` in this example code. 
```
	var a time.Time            // 0001-01-01 00:00:00 +0000 UTC
	b := a.UnixNano()          // -6795364578871345152
	c := time.Unix(0, b).UTC() // 1754-08-30 22:43:41.128654848 +0000 UTC
```

Digging a bit more into it with jsha, we noticed that the newRegistrationValid check shouldn't be checking CreatedAt in the first place. The only places where that check is used are before the correct CreatedAt value is set in sa.NewRegistration making it a meaningless check. I think we should stick with removing it.